app/app/components/AddBookToBookshelf/BookGrid.jsx

```diff
@@ -1,39 +1,72 @@
+"use client";
+import { memo, useState, useEffect, useCallback, useRef } from "react";
 import styles from "./BookGrid.module.css";
 
-const BookGrid = ({ books, onAddBook, loading }) => {
+const BookCard = memo(({ book, onAddBook }) => {
+  const [isAdding, setIsAdding] = useState(false);
+  const isMounted = useRef(true);
+
+  useEffect(() => {
+    return () => {
+      isMounted.current = false;
+    };
+  }, []);
```

Comment:
why is this needed ?

Reply:
I think I was trying to fix bad state call

Reply:
seems a little convoluted... It's probably not enough time to fix things nor it is necessary, but "bad state call" usually means something can be simplified in the component :) This is things usually take a long time to debug, so I understand and applaud the solution 👏